### PR TITLE
docs(store): add feature creator guide

### LINKED ISSUE #3183: Add guide for feature creator
Add guide for the `createFeature` function:

- What is NgRx feature?
- Example
- Advantages
- Restrictions

### PR DESCRIPTION
**Pull request by markostanimirovic**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

Closes #3183

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/store/feature-creators.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# Feature Creators
+
+## What is NgRx feature?
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/feature-creators.md:3]:
```suggestion
## What is an NgRx feature?
```

File: projects/ngrx.io/content/guide/store/feature-creators.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    })),
+    on(BooksApiActions.loadBooksSuccess, (state, { books }) => ({
+      ...state,
+      books,
+      loading: false,
+    }))
+  ),
+});
+
+const {
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/feature-creators.md:50]:
```suggestion
export const {
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: projects/ngrx.io/content/guide/store/feature-creators.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
+# Feature Creators
+
+## What is an NgRx feature?
+
+There are three main building blocks of global state management with `@ngrx/store`: actions, reducers, and selectors.
+For a particular feature state, we create a reducer for handling state transitions based on the dispatched actions
+and selectors for obtaining slices of the feature state. Also, we need to define a feature name needed to register
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/feature-creators.md:7]: I find this easier to read:

```suggestion
and selectors to obtain slices of the feature state. Also, we need to define a feature name needed to register
```

File: projects/ngrx.io/content/guide/store/feature-creators.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+the feature reducer in the NgRx store. Therefore, we can consider the NgRx feature as a group of feature name,
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/feature-creators.md:8]:
```suggestion
the feature reducer in the NgRx store. Therefore, we can consider the NgRx feature as a grouping of the feature name,
```

File: projects/ngrx.io/content/guide/store/feature-creators.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+feature reducer, and selectors for the particular feature state.
+
+## Using feature creator
+
+The `createFeature` function reduces repetitive code in selector files by generating a feature selector and child selectors
+for each feature state property. It accepts an object containing a feature name and a feature reducer as the input argument:
+
+<code-example header="books.reducer.ts">
+// `createFeature` is imported from `@ngrx/store` package
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/feature-creators.md:17]:
```suggestion
```

File: projects/ngrx.io/content/guide/store/feature-creators.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  selectLoading, // selector for `loading` property
+} = booksFeature;
+</code-example>
+
+An object created with the `createFeature` function contains a feature name, a feature reducer, a feature selector,
+and a selector for each feature state property. All generated selectors have the "select" prefix, and the feature selector has
+the "State" suffix. In this example, the name of the feature selector is `selectBooksState`, where "books" is the feature name.
+The names of the child selectors are `selectBooks` and `selectLoading`, based on the property names of the books feature state.
+
+Generated selectors can be used independently or to create other selectors:
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/feature-creators.md:64]:
```suggestion
The generated selectors can be used independently or to create other selectors:
```

File: projects/ngrx.io/content/guide/store/feature-creators.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+import { booksFeature } from './books.reducer';
+
+export const selectBookListPageViewModel = createSelector(
+  booksFeature.selectBooks,
+  booksFeature.selectLoading,
+  (books, loading) => ({ books, loading })
+);
+</code-example>
+
+Registering the feature reducer in the store can be done by passing the entire feature object to the `StoreModule.forFeature` method:
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/feature-creators.md:77]: What about adding a header here?
```suggestion
## Feature Registration 

Registering the feature reducer in the store can be done by passing the entire feature object to the `StoreModule.forFeature` method:
```

**Review by timdeschryver — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**